src/cluster_asm.c

```diff
@@ -298,23 +380,26 @@ static void clusterCommandMigrationStatus(client *c) {
     listIter li;
     listNode *ln;
 
-    addReplyArrayLen(c, listLength(server.cluster->asm_links));
+    addReplyArrayLen(c, listLength(server.cluster->asm_tasks));
 
-    listRewind(server.cluster->asm_links, &li);
+    listRewind(server.cluster->asm_tasks, &li);
     while ((ln = listNext(&li)) != NULL) {
-        asmTask *link = ln->value;
+        asmTask *task = listNodeValue(ln);
+        sds slots_range_str = createSlotRangesStr(task->slot_ranges);
 
         addReplyMapLen(c, 5);
         addReplyBulkCString(c, "slots_range");
-        addReplyBulkSds(c, createSlotRangesStr(link->slot_ranges));
+        addReplyBulkSds(c, slots_range_str);
         addReplyBulkCString(c, "source");
-        addReplyBulkCBuffer(c, link->source, CLUSTER_NAMELEN);
+        addReplyBulkCBuffer(c, task->source, CLUSTER_NAMELEN);
         addReplyBulkCString(c, "dest");
-        addReplyBulkCBuffer(c, link->dest, CLUSTER_NAMELEN);
+        addReplyBulkCBuffer(c, task->dest, CLUSTER_NAMELEN);
         addReplyBulkCString(c, "operation");
-        addReplyBulkCString(c, link->operation == ASM_IMPORT ? "importing" : "migrating");
+        addReplyBulkCString(c, task->operation == ASM_IMPORT ? "importing" : "migrating");
         addReplyBulkCString(c, "state");
-        addReplyBulkCString(c, link->state);
+        addReplyBulkCString(c, asmTaskStateToString(task->state));
+
+        sdsfree(slots_range_str);
```

Comment:
isn't this double free now? because addReplyBulkSds() will free it, no?

Reply:
when i fixed my bug, I thought there was a problem here, too.